kotlinx-coroutines-core/jvm/src/scheduling/Dispatcher.kt

```diff
@@ -38,6 +46,13 @@ private object UnlimitedIoScheduler : CoroutineDispatcher() {
     override fun dispatch(context: CoroutineContext, block: Runnable) {
         DefaultScheduler.dispatchWithContext(block, BlockingContext, false)
     }
+
+    @ExperimentalCoroutinesApi
+    override fun limitedParallelism(parallelism: Int): CoroutineDispatcher {
+        parallelism.checkParallelism()
+        if (parallelism == Int.MAX_VALUE) return this
```

Comment:
```suggestion
        if (parallelism >= MAX_POOL_SIZE) return this
```
Symmetrical and also a bit more general, in case someone overrides `MAX_POOL_SIZE` to a lower value.

Reply:
I can rebase the commit to include MAX_POOL_SIZE into original commit description instead of Int.MAX_VALUE if you wish

Reply:
We squash PRs into one commit and manually go through commit messages before merging, so it's not a big deal.